src/sagemaker/workflow/_repack_model.py

```diff
@@ -51,17 +55,59 @@
         shutil.copy2(model_path, local_path)
         src_dir = os.path.join(tmp, "src")
         # create the "code" directory which will contain the inference script
-        os.makedirs(os.path.join(src_dir, "code"))
+        code_dir = os.path.join(src_dir, "code")
+        os.makedirs(code_dir)
         # extract the contents of the previous training job's model archive to the "src"
         # directory of this training job
         with tarfile.open(name=local_path, mode="r:gz") as tf:
             tf.extractall(path=src_dir)
 
-        # generate a path to the custom inference script
-        entry_point = os.path.join("/opt/ml/code", args.inference_script)
-        # copy the custom inference script to the "src" dir
-        shutil.copy2(entry_point, os.path.join(src_dir, "code", args.inference_script))
+        # copy the custom inference script to code/
+        entry_point = os.path.join("/opt/ml/code", inference_script)
+        shutil.copy2(entry_point, os.path.join(src_dir, "code", inference_script))
+
+        # copy source_dir to code/
+        if source_dir:
+            actual_source_dir_path = os.path.join("/opt/ml/code", source_dir)
+            if os.path.exists(actual_source_dir_path):
+                for item in os.listdir(actual_source_dir_path):
+                    s = os.path.join(actual_source_dir_path, item)
+                    d = os.path.join(os.path.join(code_dir, source_dir), item)
+                    if os.path.isdir(s):
+                        shutil.copytree(s, d)
+                    else:
+                        shutil.copy2(s, d)
+
+        # copy any dependencies to code/lib/
+        if dependencies:
+            for dependency in dependencies.split(" "):
+                actual_dependency_path = os.path.join("/opt/ml/code", dependency)
+                lib_dir = os.path.join(code_dir, "lib")
+                if not os.path.exists(lib_dir):
+                    os.mkdir(lib_dir)
+                if os.path.isdir(actual_dependency_path):
+                    shutil.copytree(
+                        actual_dependency_path,
+                        os.path.join(lib_dir, os.path.basename(actual_dependency_path)),
```

Comment:
why is `os.path.join` needed here but not on line 77?

Reply:
`os.path.join` is being used for both calls, but on line 77 it's being done in the assignment of the `s` and `d` variables